libcloudforensics/providers/aws/forensics.py

```diff
@@ -370,44 +381,66 @@ def CopyEBSSnapshotToS3(
     if result['CreationDate'] > date:
       ami_id = result['ImageId']
       date = result['CreationDate']
-  assert ami_id  # Mypy: assert that ami is not None
+  if not ami_id:
+    raise errors.ResourceCreationError(
+      'Could not fnd suitable AMI for instance creation', __name__)
+
+  # Instance role creation has a propagation delay between creating in IAM and
+  # being usable in EC2.
+  if prof_created:
+    sleep(20)
 
   # start the VM
   logger.info('Starting copy instance')
   aws_account.ec2.GetOrCreateVm(
-    'ebsCopy',
+    'ebsCopy-{0:d}'.format(random.randint(10**(9),(10**10)-1)),
     10,
     ami_id,
     4,
     subnet_id=subnet_id,
     security_group_id=security_group_id,
     userdata=startup_script,
     instance_profile=instance_profile_arn,
-    terminate_on_shutdown=True
+    terminate_on_shutdown=True,
+    wait_for_health_checks=False
   )
-  logger.info('Instance creation completed')
-  logger.info('Checking for destination files with exponential backoff')
+  logger.info('Checking for output files with exponential backoff')
 
   wait = 10
   tries = 6 # 10.5 minutes
   success = False
 
   while tries:
     tries -= 1
-    logger.info("Waiting {0:d} seconds".format(wait))
+    logger.info('Waiting {0:d} seconds'.format(wait))
     sleep(wait)
     wait *= 2
 
     # pylint: disable=line-too-long
-    if aws_account.s3.CheckForObject(path_components[0], "{0:s}/{1:s}.bin".format(path_components[1], snapshot_id)) and \
-        aws_account.s3.CheckForObject(path_components[0], "{0:s}/{1:s}.sha256".format(path_components[1], snapshot_id)):
+    if aws_account.s3.CheckForObject(path_components[0], '{0:s}/{1:s}/image.bin'.format(path_components[1], snapshot_id)) and \
```

Comment:
Please split this into variables, e.g. one for the string prefix (`{0:s}/{1:s}`), and another for `path_components[0]`  (what's `path_components[0]` anyways?) then you can do:

```python
foo = path_components[0]
prefix = '{0:s}/{1:s}/'.format(path_components[1], snapshot_id)
files = ['image.bin', 'log.txt', 'hlog.txt', 'mlog.tyxt']
checks = [aws_acccount.s3.CheckForObject(foo, prefix + file) for file in files]
if all(checks):
  success = True
  break
```